src/main/python/ayab/ayab.py

```diff
@@ -65,18 +75,51 @@
     QtWidgets.QApplication.setAttribute(Qt.AA_UseHighDpiPixmaps, True)
 # Remove Help Button
 if hasattr(Qt, 'AA_DisableWindowContextHelpButton'):
-    QtWidgets.QApplication.setAttribute(Qt.AA_DisableWindowContextHelpButton,
-                                        True)
+    QtWidgets.QApplication.setAttribute(Qt.AA_DisableWindowContextHelpButton, True)
+
+
+def label(text):
+    table = "<table><tbody><tr height='12'><td style='font-weight: normal;'>" + text + "</td></tr></tbody></table>"
+    label = QtWidgets.QLabel()
+    label.setText(table)
+    label.setTextFormat(Qt.RichText)
+    return label
+
+def stitch(color, symbol, bit, alt_color=None):
+    table = "<table style='font-weight: normal;'><tbody><tr height='12'><td width='12' align='center' "
+    if bit:
+        table = table + "style='border: 1 solid black; color: #{:06x}; background-color: #{:06x};'>".format(contrast_color(color), color) + chr(symbol)
+    elif alt_color is not None:
+        table = table + "style='border: 1 solid black; color: #{:06x}; background-color: #{:06x};'>".format(contrast_color(alt_color), alt_color) + chr(symbol)
+    else:
+        table = table + "style='border: 1 dotted black;'>"
+    table = table + "</td></tr></tbody></table>"
+    label = QtWidgets.QLabel()
+    label.setText(table)
+    label.setTextFormat(Qt.RichText)
+    return label
+
+def rgb2array(color):
+    return np.array([color // 0x1000, (color // 0x100) & 0xFF, color & 0xFF])
+
+def greyscale(rgb):
+    return np.dot([.299, .587, .114], rgb)
+
+def contrast_color(color):
+    return [0x000000,0xFFFFFF][greyscale(rgb2array(color)) < 0x80]
+
```

Comment:
Probably worth to put these functions as methods into the ayab_progress class.

Reply:
Yes